Tests/AuthFoundationTests/ApiRetryTests.swift

```diff
@@ -0,0 +1,281 @@
+//
+// Copyright (c) 2022-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import XCTest
+@testable import AuthFoundation
+@testable import TestCommon
+
+class ApiRetryTests: XCTestCase {
```

Comment:
Since many of these tests use the same expectation (e.g. identical response headers and status code) you might be able to simplify this by reusing the same expectation.  You can either create a function to create this expectation `n` number of times, or you can rearrange some of the logic in `URLSessionMock.swift` to allow you to more simply create a `URLSessionMock.Call` object.  For example, in `URLSessionMock.expect(_:data:statusCode:contentType:error:)`, you can move the creation of the response to a convenience initializer on the `Call` type, allowing you to create a `let` value for the call.

Either way, I think this code can be cleaned up a bit.

Reply:
Good callout. I see what you're saying. I can use a function n times to represent it.

Reply:
Good callout. I've updated this.